src/standalone/Snow/snow_parameterizations.jl

```diff
@@ -363,3 +370,202 @@ function energy_from_T_and_swe(S::FT, T::FT, parameters) where {FT}
     end
 
 end
+
+"""
+    compat_density(W_i::FT, sliq::FT, ρ::FT, Δt_t::FT, Tsnow::FT, density::Anderson1976) where {FT}
+Returns the new compressed density ρ_new from the input ρ under the compaction model defined
+by Snow17 for a given `Anderson1976` parameterization configuration.
+"""
+function compat_density(
+    W_i::FT,
+    sliq::FT,
+    ρ::FT,
+    Δt_t::FT,
+    Tsnow::FT,
+    density::Anderson1976,
+)::FT where {FT}
+    c5 = (sliq > FT(0)) ? FT(2) : density.c5
+    cx = (ρ > density.ρ_d) ? density.cx : FT(0)
+
+    B =
+        FT(100) *
+        W_i *
+        density.c1 *
+        Δt_t *
+        exp(FT(0.08) * Tsnow - density.c2 * ρ)
+    factor = (W_i > FT(0.001)) ? (exp(B) - FT(1)) / B : FT(1)
+    A = exp(
+        density.c3 *
+        c5 *
+        Δt_t *
+        exp(density.c4 * Tsnow - cx * (ρ - density.ρ_d)),
+    )
+    factor = factor * A
+
+    ρ_new = minimum([ρ * factor, FT(0.45)])
+    ρ_new = (ρ_new < FT(0.05)) ? ρ : ρ_new
+    return ρ_new
+end
+
+"""
+    newsnow_dens(air_temp::FT)::FT where {FT}
+Helper function for dzdt under an `Anderson1976` density parameterization.
+A model for estimating the (as a fraction of liquid water density)
+density of newly fallen snow, given air temperature. Uses the model implemented
+in Snow17.
+"""
+function newsnow_dens(air_temp::FT)::FT where {FT}
+    if air_temp < FT(-15.0)
+        return FT(0.05)
+    else
+        return FT(0.05) + FT(0.0017) * FT((air_temp + FT(15.0))^1.5)
+    end
+end
+
+"""
+    newsnow_dens(air_temp::FT)::FT where {FT}
```

Comment:
can update doc string for the function below

Reply:
Typically the doc string should first describe what the function does, and then provide details:

"Estimates the temperature of newly fallen snow as a function of air temperature, according to the Snow17 model.

Used in computing the time derivative of snow depth under the `Anderson1976` density parameterization.
"